File: TAGS/GeoServices v2.0/Ejemplos/VB .Net/EjemploGeoApp/EjemploGeoApp/EjemploWorkingComponent.vb
Imports ESRI.ArcGIS.Geodatabase

Class EjemploWorkingComponent
    '' Extiendo de WorkingComponent
    Inherits GeoServices.Worker.WorkingComponent

    Protected Overrides Sub doExecute()
        '' Aca va el flujo de ejecución

        '' Muestro todos los Feature Class por pantalla
        Me.ListarFeatureClasses()
    End Sub

    Private Sub ListarFeatureClasses()
        '' Por cada Feature Class en el SDE
        For Each pFClass As IFeatureClass In New GeoServices.SDE.FeatureClassesGateway().GetAll()
            '' Muestro el nombre por pantalla
            Console.WriteLine(pFClass.AliasName)
        Next
    End Sub

End Class

File: TAGS/GeoServices v1.5/Ejemplos/C#/EjemploGeoApp/EjemploGeoApp/EjemploWorkingComponent.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using ESRI.ArcGIS.Geodatabase;

namespace EjemploGeoApp
{
    // Extiendo de WorkingComponent
    class EjemploWorkingComponent : GeoServices.Worker.WorkingComponent
    {
        protected override void doExecute()
        {
            // Aca va el flujo de ejecución

            // Muestro todos los Feature Class por pantalla
            this.ListarFeatureClasses();
        }

        private void ListarFeatureClasses()
        {
            // Por cada Feature Class en el SDE
            foreach (IFeatureClass pFClass in new GeoServices.XML.XMLGeoGetter().GetFeatureClasses())
                // Muestro el nombre por pantalla
                Console.WriteLine(pFClass.AliasName);
        }
    }
}
